docs/en/upload.md

```diff
@@ -56,21 +56,63 @@ function startup()
     end
 end
 
+-- Define WiFi station event callbacks 
+wifi_connect_event = function(T) 
+  print("Connection to AP("..T.SSID..") established!")
+  print("Waiting for ip...") 
+end
+
+wifi_got_ip_event = function(T) 
+  print("Wifi connection is ready! IP address is: "..T.IP)
+  print("Startup will resume momentarily, you have 3 seconds to abort.")
+  print("Waiting...") 
+  tmr.create():alarm(3000, tmr.ALARM_SINGLE, startup)
+end
+
+wifi_disconnect_event = function(T)
+  if T.reason == wifi.eventmon.reason.ASSOC_LEAVE then 
+    --the station has disassociated from a previously connected AP
+    return 
+  end
+  -- total_tries: how many times the station will attempt to connect to the AP.
+  local total_tries = 3
```

Comment:
As a default value I would increase this significantly (or drop it altogether) to accommodate for intermittent  AP reboots.